packages/diagrams/frontend/sirius-components-diagrams-reactflow/src/renderer/palette/tool-section/ToolSection.tsx

```diff
@@ -44,75 +44,96 @@ const useToolSectionStyles = makeStyles(() => ({
 const isSingleClickOnDiagramElementTool = (tool: GQLTool): tool is GQLSingleClickOnDiagramElementTool =>
   tool.__typename === 'SingleClickOnDiagramElementTool';
 
-export const ToolSection = ({ toolSection, onToolClick }: ToolSectionProps) => {
-  const tools = toolSection.tools.filter(isSingleClickOnDiagramElementTool);
-  const initialState = {
-    expanded: false,
-  };
-  const [state, setState] = useState<ToolSectionState>(initialState);
-  const { expanded } = state;
+export const ToolSection = forwardRef<ToolSectionForwardRefValues, ToolSectionProps>(
+  ({ toolSection, onToolClick, id, handleCloseAllOthersToolSection }, ref) => {
+    useImperativeHandle(ref, () => ({
+      handleClose() {
+        setState((prevState) => {
+          return {
+            ...prevState,
+            expanded: false,
+          };
+        });
+      },
+      getKey() {
+        return id;
+      },
+    }));
 
-  const classes = useToolSectionStyles();
+    const tools = toolSection.tools.filter(isSingleClickOnDiagramElementTool);
+    const initialState = {
+      expanded: false,
+    };
+    const [state, setState] = useState<ToolSectionState>(initialState);
+    const { expanded } = state;
 
-  const onActiveTool = useCallback(
-    (tool) => {
-      onToolClick(tool);
-      setState(() => {
+    const classes = useToolSectionStyles();
+
+    const onActiveTool = useCallback(
+      (tool) => {
+        onToolClick(tool);
+        setState((prevState) => {
+          return {
+            ...prevState,
```

Comment:
The ToolSection's state still only contains the `expanded` attribute as before. Maybe it's a leftover from a previous attempt where you added fields to it, but this merging of the `prevState` (here and line 89 below) does not seem necessary.